src/main/java/ecommerce/dao/impl/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,183 @@
+package ecommerce.dao.impl.jdbc;
+
+import ecommerce.dao.interfaces.UserDao;
+import ecommerce.exceptions.DataProcessingException;
+import ecommerce.lib.Dao;
+import ecommerce.model.Role;
+import ecommerce.model.User;
+import ecommerce.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE login = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    User user = getUserFromResultSet(resultSet);
+                    user.setRoles(getRolesByUserId(user.getId(), connection));
+                    return Optional.of(user);
+                }
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get user with login " + login, e);
+        }
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (name, login, password) VALUES (?, ?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            try (ResultSet resultSet = statement.getGeneratedKeys()) {
+                resultSet.next();
+                user.setId(resultSet.getLong(1));
+                addRolesToUser(user, connection);
+                return user;
```

Comment:
does user that you return has ids in roles, if yes when did you set them ?

Reply:
he does
they were setted in addRolesToUser method